### PR TITLE
qemu.tests：add case for checking usb device

### PR DESCRIPTION
**Pull request by maxujun**
qemu.tests：add case for checking usb device
Deps on [631](https://github.com/avocado-framework/avocado-vt/pull/631)
id:1324308 1324310 1324320

### CODE REVIEW

**Comment by suqinhuang** [comment]
commit log pls

File: qemu/tests/usbdev_check.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    :param env: Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    check = qemu.UsbDevTest(test, params, env)
+    check.check_usb_dev(params.get("usb_type"),
+                        params.get("info_usb_name"),
+                        params.get("vp_id"),
+                        params.get("vendor"),
+                        params.get("product"),
+                        vm)
```
Comment (mazhang) [inline qemu/tests/usbdev_check.py]: Better to add error handle on check point.

File: qemu/tests/cfg/usbdev.cfg [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            usb_max_port_usbtest = 6
+        - usb-ehci:
+            usb_type_usbtest = usb-ehci
+            usb_controller = ehci
+            usb_max_port_usbtest = 6
+            no usb_ccid
+        - nec-xhci:
+            no RHEL.3
+            no RHEL.4
+            no RHEL.5
```
Comment (CongLi) [inline qemu/tests/cfg/usbdev.cfg]: The 3 lines above can write into on line.
like: no RHEL.3, RHEL.4...

File: qemu/tests/cfg/usbdev.cfg [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            udevs += " d0"
+            start_vm = no
+    # usb tests
+    variants:
+        - check:
+            type = usbdev_check
+        - hotplug:
+            no RHEL.3
+            no RHEL.4
+            no RHEL.5
```
Comment (CongLi) [inline qemu/tests/cfg/usbdev.cfg]: Same as above

File: qemu/tests/usbdev_check.py (hunk 1 above)
Comment (maxujun) [inline qemu/tests/usbdev_check.py]: @mazhang error handle include in the UsbDevTest class

**Comment by maxujun** [comment]
@CongLi Have updated,thanks for comments

File: qemu/tests/cfg/usbdev.cfg [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            product = "QEMU USB HARDDRIVE"
+            no piix3-uhci piix4-uhci ich9-uhci
+            max_usb_dev:
+                usb_devices = " tablet1 hub1"
+            udevs += " d0"
+            start_vm = no
+    # usb tests
+    variants:
+        - check:
+            type = usbdev_check
```
Comment (Zhengtong) [inline qemu/tests/cfg/usbdev.cfg]: Failed on ppc 

usbdev.check.usb_hub.without_usb_hub.ich9-ehci
05:52:39 ERROR|   File "/root/test_home/liuzt/source_code/staf-kvm-devel/workspace/avocado-vt/virttest/utils_test/qemu.py", line 1890, in check_usb_dev
05:52:39 ERROR|     raise exceptions.TestFail("Could not find item '%s' in guest" % item)
05:52:39 ERROR| TestFail: Could not find item 'QEMU USB Hub' in guest
05:52:39 ERROR|
05:52:39 ERROR| FAIL 18-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.usbdev.check.usb_hub.without_usb_hub.ich9-ehci -> TestFail: Could not find item 'QEMU USB Hub' in guest
Checked insides guest. The name of usb_hub is not existed indeed. So it is possible to add the exception?

Comment (maxujun) [inline qemu/tests/cfg/usbdev.cfg]: @Zhengtong Have updated,thanks for comments

File: qemu/tests/cfg/usbdev.cfg [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            type = usbdev_migration
+        - function:
+            type = usbdev_func
+            variants:
+                - io_without_fs:
+                    only usb_storage usb_bot
+                    function = io_without_fs
+                - io_with_fs:
+                    only usb_storage usb_bot
+                    function = io_with_fs
```
Comment (Zhengtong) [inline qemu/tests/cfg/usbdev.cfg]: For the type "usbdev_hotplug", "usbdev_migration", "usbdev_func", I didn't find the related test modules in tp-qemu repo, will you add these test modules later? or they stays at some where else I didn't notice?

Comment (maxujun) [inline qemu/tests/cfg/usbdev.cfg]: @Zhengtong they have been added as single patch,and you can find them at:
https://github.com/autotest/tp-qemu/pull/710
https://github.com/autotest/tp-qemu/pull/709
https://github.com/autotest/tp-qemu/pull/708

**Comment by Zhengtong** [comment]
Acked  <zhengtli@redhat.com>

**Comment by CongLi** [comment]
Close it as @maxujun has no time to handle it.

**Pull request closed without merging**